Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -227,84 +233,98 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
+                      then do
+                        CQDGR.deactivateDriverGoHomeRequest booking.providerId driverId DDGR.SUCCESS ghInfo (Just True)
+                        return $ Just True
+                      else do
+                        CQDGR.resetDriverGoHomeRequest booking.providerId driverId goHomeConfig ghInfo
+                        return $ Just False
+                  Nothing -> return Nothing
+          else return Nothing
+      _ -> return Nothing
 
   whenWithLocationUpdatesLock driverId $ do
-    now <- getCurrentTime
     thresholdConfig <- findConfig >>= fromMaybeM (InternalError "TransportConfigNotFound")
+    now <- getCurrentTime
     (chargeableDistance, finalFare, mbUpdatedFareParams, ride, pickupDropOutsideOfThreshold, distanceCalculationFailed) <-
-      case req of
-        CronJobReq _ -> do
-          logTagInfo "cron job -> endRide : " "Do not call snapToRoad, return estimates as final values."
-          (chargeableDistance, finalFare, mbUpdatedFareParams) <- recalculateFareForDistance handle booking rideOld booking.estimatedDistance
-          pure (chargeableDistance, finalFare, mbUpdatedFareParams, rideOld, Nothing, Nothing)
-        _ -> do
-          -- here we update the current ride, so below we fetch the updated version
-          pickupDropOutsideOfThreshold <- isPickupDropOutsideOfThreshold booking rideOld tripEndPoint thresholdConfig
-          enableLocationTrackingService <- asks (.enableLocationTrackingService)
-          tripEndPoints <-
-            if enableLocationTrackingService
-              then do
-                res <- LF.rideEnd rideId tripEndPoint.lat tripEndPoint.lon booking.providerId driverId
-                pure $ toList res.loc
-              else pure [tripEndPoint]
-          whenJust (nonEmpty tripEndPoints) \tripEndPoints' -> do
-            withTimeAPI "endRide" "finalDistanceCalculation" $ finalDistanceCalculation rideOld.id driverId tripEndPoints' booking.estimatedDistance pickupDropOutsideOfThreshold
-
-          ride <- findRideById (cast rideId) >>= fromMaybeM (RideDoesNotExist rideId.getId)
-
-          distanceCalculationFailed <- withTimeAPI "endRide" "isDistanceCalculationFailed" $ isDistanceCalculationFailed driverId
-          when distanceCalculationFailed $ logWarning $ "Failed to calculate distance for this ride: " <> ride.id.getId
-          (chargeableDistance, finalFare, mbUpdatedFareParams) <-
-            if distanceCalculationFailed
-              then calculateFinalValuesForFailedDistanceCalculations handle booking ride tripEndPoint pickupDropOutsideOfThreshold thresholdConfig
-              else calculateFinalValuesForCorrectDistanceCalculations handle booking ride booking.maxEstimatedDistance pickupDropOutsideOfThreshold thresholdConfig
-          pure (chargeableDistance, finalFare, mbUpdatedFareParams, ride, Just pickupDropOutsideOfThreshold, Just distanceCalculationFailed)
+      case odometerEndReading of
```

Comment:
We should not just say if endRide contains `odometerEndReading` then it is Rental end ride. It should have a proper request type for endRide with Rental type.

Reply:
Yes, already have mentioned this. Once protocol changes are done, we can do that
<img width="426" alt="Screenshot 2023-10-17 at 12 17 08 PM" src="https://github.com/nammayatri/nammayatri/assets/33424495/20376b4d-e2a5-4c03-a9be-6c4cb376bdf8">